common/message_ID_domain.py

```diff
@@ -133,52 +137,41 @@ def orgGroups(self, sender, mID):
             self.domain2domainPairing[(newmID, afterAT)] = False
             return False
 
-    def create_sender_profile(self, numSamples):
-        emails_with_sender = 0
-        no_messageIDDomain = 0
-        new_format_found = 0
-        self.sender_profile = {}
-        # for msg in self.inbox:
-        for i in range(numSamples):
-            msg = self.inbox[i]
-            # if "List-Unsubscribe" in msg.keys():
-            #     continue
-            sender = self.extract_from(msg)
-            if sender:
-                emails_with_sender += 1
-                mID = self.get_endMessageIDDomain(self.get_messageIDDomain(msg))
-                if mID == None:
-                    no_messageIDDomain += 1
-                    if sender not in self.sender_profile:
-                        self.sender_profile[sender] = set([])
-                    self.sender_profile[sender].add("None")
+    def update_sender_profile(self, msg):
+        # if "List-Unsubscribe" in msg.keys():
+        #     continue
+        sender = self.extract_from(msg)
+        if sender:
+            self.emails_with_sender += 1
+            mID = self.get_endMessageIDDomain(self.get_messageIDDomain(msg))
+            if mID == None:
+                self.no_messageIDDomain += 1
+                if sender not in self.sender_profile:
+                    self.sender_profile[sender] = set([])
+                self.sender_profile[sender].add("None")
+            else:
+                wasntInSenderProfile = False
+                if sender in self.sender_profile:
+                    if mID not in self.sender_profile[sender]:
+                        if self.toFlag(sender, mID):
+                            wasntInSenderProfile = True
+                        self.sender_profile[sender].add(mID)
                 else:
-                    wasntInSenderProfile = False
-                    if sender in self.sender_profile:
-                        if mID not in self.sender_profile[sender]:
-                            if self.toFlag(sender, mID):
-                                wasntInSenderProfile = True
-                            self.sender_profile[sender].add(mID)
+                    self.sender_profile[sender] = set([mID])
+                email_domain = self.getEmailDomain(sender)
+                if email_domain not in self.GLOBAL_SET:
```

Comment:
minor nit, but when accessing/modifying class variables, I prefer using `MessageIdDomainDetector.GLOBAL_SET` instead of `self.GLOBAL_SET`, since it's possible that in the future, we accidentally instantiate an instance variable called `self.GLOBAL_SET` and we'll be modifying that one instead. (We can change this later though, since you weren't the original author of this portion of the code)